### ISSUE TITLE
Accessibility bug Article ergonomy

### ISSUE THREAD

**Issue by michalradacz** [Jermolene/TiddlyWiki5#7629]
### Describe the bug

Please modify storyriver template to add ARTICLE elements. Enclose every whole tiddler to ARTICLE element. It may rapidly improving navigation with screenreaders end ergonomy for declaring tiddlers in displayed storyriver

### TiddlyWiki Configuration

- Version all
- Saving mechanism all
- Plugins installed irelevant

### Desktop (please complete the following information):

- OS: all
- Browser all

**Comment by michalradacz** [comment]
Ehm, i tested modify it myself but it is more difficult to me find all needed occurences.

**Comment by pmario** [comment]
If you open the [ViewTemplate](https://tiddlywiki.com/#%24%3A%2Fcore%2Fui%2FViewTemplate) at tiddlywiki.com and use the following code to replace the content of the ViewTemplate tiddler, you should be able to test, if that would make sense with a screenreader. 

Which screenreader software do you use?

```
\whitespace trim
\define folded-state()
$:/state/folded/$(currentTiddler)$
\end
\define cancel-delete-tiddler-actions(message) <$action-sendmessage $message="tm-$message$-tiddler"/>
\import [all[shadows+tiddlers]tag[$:/tags/Macro/View]!is[draft]] [all[shadows+tiddlers]tag[$:/tags/Global/View]!is[draft]]
<$vars storyTiddler=<<currentTiddler>> tiddlerInfoState=<<qualify "$:/state/popup/tiddler-info">>>
<article data-tiddler-title=<<currentTiddler>> data-tags={{!!tags}} class={{{ [all[shadows+tiddlers]tag[$:/tags/ClassFilters/TiddlerTemplate]!is[draft]] :map:flat[subfilter{!!text}] tc-tiddler-frame tc-tiddler-view-frame [<currentTiddler>is[tiddler]then[tc-tiddler-exists]] [<currentTiddler>is[missing]!is[shadow]then[tc-tiddler-missing]] [<currentTiddler>is[shadow]then[tc-tiddler-exists tc-tiddler-shadow]] [<currentTiddler>is[shadow]is[tiddler]then[tc-tiddler-overridden-shadow]] [<currentTiddler>is[system]then[tc-tiddler-system]] [{!!class}] [<currentTiddler>tags[]encodeuricomponent[]addprefix[tc-tagged-]] +[join[ ]] }}} role="article">
<$list filter="[all[shadows+tiddlers]tag[$:/tags/ViewTemplate]!is[draft]]" variable="listItem">
<$transclude tiddler=<<listItem>>/>
</$list>
</article>
</$vars>
```

**Comment by pmario** [comment]
The ViewTemplate tiddler from my first post is only 1 template. It's used to render "content" tiddlers. So that's probably 90% of tiddlers. We would need to modify some more templates to catch all the possibilities. But IMO it should be a start.

**Comment by Jermolene** [comment]
Welcomei @michalradacz we love feedback that helps us make TiddlyWiki better for users of assistive technology. From a backwards compatibility perspective t's too late to change the HTML elements used in the main templates, would it be possible to use the `aria-role` attribute instead?

**Comment by Leilei332** [comment]
@saqimtiaz @pmario IMO this issue can be closed, since its purpose is intended to improve accessibility. Tiddlywiki do adds `article` roles to the tiddlers now.

**Comment by michalradacz** [comment]
I agree

16. 1. 2026 v 11:22, XLBilly ***@***.***>:

[https://avatars.githubusercontent.com/u/70204468?s=20&v=4]Leilei332 left a comment (TiddlyWiki/TiddlyWiki5#7629)<https://github.com/TiddlyWiki/TiddlyWiki5/issues/7629#issuecomment-3759280025>

@saqimtiaz<https://github.com/saqimtiaz> @pmario<https://github.com/pmario> IMO this issue can be closed, since its purpose is intended to improve accessibility. Tiddlywiki do adds article roles to the tiddlers now.
